Do laundry

VincentNe opened the pull request.

VincentNe commented:
Waiting for:
the machine state to be merged with the master branch.
Authentication with the machine.

VincentNe commented:
Do laundry done, can be tested with the view I created (Webapp>machineView)

I added the lockstates and made some other small changes to state logic. Therefore I think we should review the command pattern.

**Reason build error:**
duplicate users_pkey -> I use user id 1 in my import.sql script just like you use in your tests.

VincentNe commented:
updated the branch with the master branch.

@Raul6469 
We have a problem with the command pattern -> is it possible to add a program id into the start machine command?

Raul6469 commented:
Ok, thanks for it, but I don't see what you've done here, there is a lot of things going on...
What I can just say right now is that Travis is not failing because of SQL statements, but because of compilation errors, can you start by trying to fix it please?

AmosTrask commented on src/main/java/euteam/cleanmachine/controller/MachineController.java:
```diff
@@ -1,29 +1,193 @@
+
+    @RequestMapping(path="/machine/startProgram", method = RequestMethod.POST)
+    public ResponseEntity<?> startProgram(@RequestParam("programId") long programId){
+        Machine machine = getMachineByAuthentication();
```
Use a service for the business logic.

VincentNe replied:
okay, I was thinking the same but wasn't sure

AmosTrask replied:
👍

AmosTrask commented on src/main/java/euteam/cleanmachine/service/MachineService.java:
```diff
@@ -1,37 +1,57 @@
-
-        return machineDao.save(machine);
-    }
-}
```
Do you know why the diff looks like this ? ^^^Kinda annoying to see the changes.

VincentNe replied:
idk

VincentNe commented:
> Ok, thanks for it, but I don't see what you've done here, there is a lot of things going on...
> What I can just say right now is that Travis is not failing because of SQL statements, but because of compilation errors, can you start by trying to fix it please?

Oke fixed the compilation errors -> happened during update with master

Like @AmosTrask mentioned, I need to refactor some code in the controller and place it in the service

VincentNe commented:
**update**
- Removed duplicate identifier in machine class
        Removed Long id in machine class .
        String identifier is know annotated with Id and auto generated with the UUID class.
- Refactor controller
        Moved the business logic from the controller to the service.
- Fixed tests
        Who couldn't do 4+4??? I can tell you one thing .. it's not 11... 🥇

AmosTrask commented on src/main/java/euteam/cleanmachine/model/facility/Program.java:
```diff
@@ -6,18 +6,16 @@
 
-    public Program() {
-        this.cost = 3;
-    }
```
Here is the missing 3.
3+4+4 = 11

VincentNe replied:
Oo I see, sorry removed it because I didn't see why it should be 3

VincentNe replied:
Oo I see, sorry removed it because I didn't see why it should be 3

Raul6469 commented on src/main/java/euteam/cleanmachine/model/facility/Machine.java:
```diff
@@ -1,83 +1,138 @@
-    private Long id;
+    @GeneratedValue(generator = "uuid")
+    @GenericGenerator(name = "uuid", strategy = "uuid")
+    private String identifier;
```
I thought of the identifier as the machine serial number, the one provided by the manufacturer

VincentNe replied:
Oke keep it like this, except we don't generate the id but give it in the constructor? I assume serial number is unique?

Raul6469 commented on src/main/java/euteam/cleanmachine/model/facility/Machine.java:
```diff
@@ -1,83 +1,138 @@
+    public boolean containsProgram(Long programId){
+        for (Program program:programs
+             ) {
+            if(program.getId()==programId)return true;
```
Looks like when you try to start the woll program through the web interface it tells that the program doesn't exist, so this change fixes the bug
```suggestion
            if(program.getId().equals(programId)) return true;
```

VincentNe commented on src/main/java/euteam/cleanmachine/security/WebSecurityConfig.java:
```diff
@@ -65,6 +65,7 @@ protected void configure(HttpSecurity httpSecurity) throws Exception {
             .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS).and()
             .authorizeRequests()
             .antMatchers(HttpMethod.OPTIONS,"/**").permitAll()
+            .antMatchers("/machine").permitAll()
```
I don't think so, I forgot to remove it. 
```suggestion
            
```

VincentNe commented on src/main/java/euteam/cleanmachine/commands/Start.java:
```diff
@@ -6,6 +6,7 @@
     @Override
     public void execute(User user, Machine machine) {
-        machine.startMachine(user.getId());
+        //TODO FIX command so we can add programID as parameter
```
we don't always need a user when either so maybe, we actually only user the userID so maybe. 

Would this be an option?
 public void execute(Long userId, Long programId , Machine machine) {}

pass null is programid needs to be send.

Another option would be to create an Object e.g 'ExecuteParam' with userId and programId.
 public void execute(ExecuteParam params, Machine machine) {}

VincentNe commented on src/main/java/euteam/cleanmachine/model/facility/Machine.java:
```diff
@@ -1,83 +1,138 @@
+          loggedInUserID=  state.getUserId();
+        }else if(state instanceof RunningState){
+            RunningState state = (RunningState) getState();
+            loggedInUserID=  state.getUserId();
```
yes, will refactor this

VincentNe replied:
should be fixed in the new commit
